### PR TITLE
reset message_b after second GetDigest.

### PR DESCRIPTION
**Pull request by qizhangz**
Fix: #916 

some digests unit test about message_b full case was removed since the message_b is reset.

Signed-off-by: Qi Zhang <qi1.zhang@intel.com>

### CODE REVIEW

File: library/spdm_requester_lib/libspdm_req_get_digests.c [hunk 1]
```diff
@@ -196,7 +196,11 @@ libspdm_return_t libspdm_try_get_digest(void *context, uint8_t *slot_mask,
         sizeof(spdm_context->connection_info.peer_total_digest_buffer),
         spdm_response->digest, digest_size * digest_count);
 
-    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_DIGESTS;
+    if (spdm_context->connection_info.connection_state <
+        LIBSPDM_CONNECTION_STATE_AUTHENTICATED) {
```
Comment (jyao1) [inline library/spdm_requester_lib/libspdm_req_get_digests.c:200]: With this, LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE will become LIBSPDM_CONNECTION_STATE_AFTER_DIGESTS.

I dont feel good. We should maintain the LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE status as well.

Comment (jyao1) [inline library/spdm_requester_lib/libspdm_req_get_digests.c:200]: With this, LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE will become LIBSPDM_CONNECTION_STATE_AFTER_DIGESTS.

I dont feel good. We should maintain the LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE status as well.

Comment (jyao1) [inline library/spdm_requester_lib/libspdm_req_get_digests.c:200]: With this, LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE will become LIBSPDM_CONNECTION_STATE_AFTER_DIGESTS.

I dont feel good. We should maintain the LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE status as well.

File: library/spdm_responder_lib/libspdm_rsp_digests.c [hunk 2]
```diff
@@ -143,8 +143,11 @@ libspdm_return_t libspdm_get_response_digests(void *context, size_t request_size
                                                response_size, response);
     }
 
-    libspdm_set_connection_state(spdm_context,
-                                 LIBSPDM_CONNECTION_STATE_AFTER_DIGESTS);
+    if (spdm_context->connection_info.connection_state <
+        LIBSPDM_CONNECTION_STATE_AUTHENTICATED) {
```
Comment (jyao1) [inline library/spdm_responder_lib/libspdm_rsp_digests.c:147]: With this, LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE will become LIBSPDM_CONNECTION_STATE_AFTER_DIGESTS.

I dont feel good. We should maintain the LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE status as well.

Comment (jyao1) [inline library/spdm_responder_lib/libspdm_rsp_digests.c:147]: With this, LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE will become LIBSPDM_CONNECTION_STATE_AFTER_DIGESTS.

I dont feel good. We should maintain the LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE status as well.

File: library/spdm_common_lib/libspdm_com_context_data.c [hunk 3]
```diff
@@ -1045,7 +1045,6 @@ void libspdm_reset_message_buffer_via_request_code(void *context, void *session_
         }
         break;
     case SPDM_GET_DIGESTS:
-    case SPDM_GET_CERTIFICATE:
         if (spdm_context->connection_info.connection_state ==
             LIBSPDM_CONNECTION_STATE_AUTHENTICATED) {
             libspdm_reset_message_b(spdm_context);
```
Comment (jyao1) [inline library/spdm_common_lib/libspdm_com_context_data.c:1050]: I am confused. How does this work?

Comment (jyao1) [inline library/spdm_common_lib/libspdm_com_context_data.c:1050]: I am confused. How does this work?

Comment (jyao1) [inline library/spdm_common_lib/libspdm_com_context_data.c:1050]: I am confused. How does this work?

File: library/spdm_requester_lib/libspdm_req_get_digests.c (hunk 1 above)
Comment (qizhangz) [reply library/spdm_requester_lib/libspdm_req_get_digests.c:200]: When GET_DIGESTS -> GET_CERTIFICATE -> GET_DIGESTS after Authenticated, the state will be LIBSPDM_CONNECTION_STATE_AUTHENTICATED and not be changed.

File: library/spdm_common_lib/libspdm_com_context_data.c (hunk 3 above)
Comment (qizhangz) [reply library/spdm_common_lib/libspdm_com_context_data.c:1050]: When second GetDigest, the connection_state is LIBSPDM_CONNECTION_STATE_AUTHENTICATED.

**Comment by jyao1** [comment]
Have you tested GET_DIGEST, GET_CERTIFICATE, GET_DIGEST flow ?

In this flow, the state should be LIBSPDM_CONNECTION_STATE_AFTER_CERTIFICATE.

Why the code depends on LIBSPDM_CONNECTION_STATE_AUTHENTICATED?

**Review by jyao1 — APPROVED** [review]

**Review by jyao1 — CHANGES_REQUESTED** [review]
Please update libspdm_test_requester_get_digests_case24() to use get_digest/get_cert/get_digest, without context change.

**Review by jyao1 — APPROVED** [review]

**Comment by jyao1** [comment]
merge it together with emu, so that we can start test.

feel free to leave more feedback, if you find something need improve later.

**Pull request merged**